### PR TITLE
Refactor: Move `parsers` to language index

### PR DESCRIPTION
**Pull request by fisker**
Move `parsers` to language index for easier maintenance. 

(When I first  try to add a parser, I don't even know the parsers need add to `common/internal-parsers.js`.)

This also make `language-xxx/index.js` a "completed" plugin, previously `index.js` + `parser-xxxx.js` is the whole plugin (standalone build still in this way).
 

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     // in other sources:
     "src/index.js",
     "src/standalone.js",
+    "src/languages.js",
```
Comment (brodybits) [inline tsconfig.json:45]: - I think we can remove `src/common/internal-plugins.js` from the list.
- I think `src/languages.js` should move a few lines up to the section with "JavaScript sources affected by JSDoc issues in src/language-*"

**Comment by brodybits** [comment]
+1 (+100) overall

> This also make `language-xxx/index.js` a "completed" plugin

:tada:

Do we think the standalone build will be updated someday?

Any interesting effect on the external API?

**Comment by fisker** [comment]
> Do we think the standalone build will be updated someday?

Maybe with `import('./parser-xxx.mjs')`, when we support asyc api and use ESModule.

> Any interesting effect on the external API?

There should not.

File: tsconfig.json (hunk 1 above)
Comment (fisker) [reply tsconfig.json:45]: Will do.

**Pull request merged**